raft/src/main/java/org/apache/kafka/snapshot/FileRawSnapshotWriter.java

```diff
@@ -74,18 +74,19 @@ public void freeze() throws IOException {
         frozen = true;
 
         // Set readonly and ignore the result
-        if (!path.toFile().setReadOnly()) {
-            throw new IOException(String.format("Unable to set file %s as read-only", path));
+        if (!tempSnapshotPath.toFile().setReadOnly()) {
+            throw new IOException(String.format("Unable to set file (%s) as read-only", tempSnapshotPath));
         }
 
-        Path destination = Snapshots.moveRename(path, snapshotId);
-        Files.move(path, destination, StandardCopyOption.ATOMIC_MOVE);
+        Path destination = Snapshots.moveRename(tempSnapshotPath, snapshotId);
+        Files.move(tempSnapshotPath, destination, StandardCopyOption.ATOMIC_MOVE);
     }
 
     @Override
     public void close() throws IOException {
         channel.close();
-        Files.deleteIfExists(path);
+        // This is a noop if freeze was called before calling close
+        Files.deleteIfExists(tempSnapshotPath);
```

Comment:
Would it be useful to call this in a `finally`?

Reply:
Yes!